jupyterhub/auth.py

```diff
@@ -762,7 +771,15 @@ def add_system_user(self, user):
         Tested to work on FreeBSD and Linux, at least.
         """
         name = user.name
-        cmd = [arg.replace('USERNAME', name) for arg in self.add_user_cmd] + [name]
+        cmd = [arg.replace('USERNAME', name) for arg in self.add_user_cmd]
+        try:
+            uid = self.uids[name]
+            cmd += ['--uid', '%d' % uid]
+        except AttributeError:
+            pass
+        except KeyError:
+            self.log.warning("No UID for user %s" % name)
```

Comment:
Let's demote this highly probable event to debug-level:

```suggestion
            self.log.debug("No UID for user %s" % name)
```